File: Front-end/Recepcion/Recepcion/src/core/Interfaces/visitas/Ivisitantes.ts
import { ClientesEntity } from "../../Entities/clients/clients";

 export abstract class  IVisitantes{

abstract crearVisitante(visitante:ClientesEntity):Promise<ClientesEntity|null>
abstract buscarVisitante(ci:string):Promise<ClientesEntity[]|[]>;
abstract getAllVisitante():Promise<ClientesEntity[]|[]>
}

File: APP/core/Interface/visitas/Ivisitantes.py
from core.Entities.clientes.clientesEntity import ClientesEntity
from abc import ABC,abstractmethod
from config.Logs.LogsActivity import Logs
class IVisitantes(ABC):
    def __init__(self):
        Logs.WirterTask("new implement IClientes")
    
    @abstractmethod
    def crearVisitante(cliente:ClientesEntity)->ClientesEntity:
        pass
    
    @abstractmethod
    def contarVisitante()->int:
        pass
    @abstractmethod
    def buscarVisitante(ci:str)->list[ClientesEntity]:
        pass
        
    @abstractmethod
    def getAllVisitantes()->list[ClientesEntity]:
        pass